registry/storage/driver/gcs/gcs.go

```diff
@@ -113,6 +117,7 @@ type baseEmbed struct {
 // Required parameters:
 // - bucket
 func FromParameters(parameters map[string]interface{}) (storagedriver.StorageDriver, error) {
+	ctx := context.Background()
```

Comment:
This would have been a way bigger change than what this PR ships. Currently, no storage driver passes context to `FromParameters`.  IMHO, should we decide to do this it should go into a separate PR.

Reply:
Yes, not as part of this PR; the only change here _could be_ to change `context.Background()` to a `context.TODO()` to more clearly indicate "this should still be punched through".

As to whether or not they should share the same context, I started reviewing before I noticed @corhere already reviewed; he's all familiar with the ins-and-outs of contexts, so I expect sharing the context is ok (otherwise he'd _very likely_ would've called it out 😂)

Reply:
Honestly I just skimmed that part of the code. [I should have given it more scrutiny.](https://pkg.go.dev/golang.org/x/oauth2#NewClient)
> The returned [http] client is not valid beyond the lifetime of the context.

It looks like golang.org/x/oauth2 captures contexts in struct fields all over the place. `oauth2.NewClient` and `jwtConf.TokenSource` not having an error-return should have tipped me off that the passed contexts weren't just being used as cancelation signals for the constructor function calls themselves. I don't see any reasonable alternative to sharing a context, but it does mean that _the constructed StorageDriver_ is not valid beyond the lifetime of the context. That's outside of the norm, so should be documented so as not to trip up people like me who expect the context lifetime to only bound the lifetime of the function call, not the validity of its return values.